python/tests/test_pyvw.py

```diff
@@ -376,6 +376,19 @@ def test_example_features():
     ex.push_namespace(ns2)
     assert ex.pop_namespace()
 
+
+def test_get_weight_name():
+    model = vw(quiet=True)
+    model.learn("1 | a a b c |ns x")
+    assert model.get_weight_from_name("a") != 0.
```

Comment:
what does get_weight_from_name behave when you ask for a feature that you haven't used before?

Reply:
might be interesting to test out with --random_weights too

Reply:
> how does get_weight_from_name behave when you ask for a feature that you haven't used before?

It returns `0.`  which is default initial value for feature name that have not appeared (yet) in the learning phase.

Reply:
> might be interesting to test out with --random_weights too

with `--random_weight` I am afraid we **won't be able to distinguish the weight initialization process from the weight update process**. To put differently, in the case of random weight initialization:
- every weight will be non 0 (except in the extremely rare case where the random weight is 0), which makes the test on nullity useless
- the equality between two weights  (for 2 features that appear in the same context such as  "b" and "c") will fail  since these 2 weights will still be different after 1 update. They might converge asymptotically (i.e after repeated updates) but that's another story.